src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -32,6 +33,20 @@ private struct AuthenticatedKeys
         private DecryptionDelegate DecryptFunction;
         private EncryptionDelegate EncryptFunction;
         private const string _className = "Microsoft.IdentityModel.Tokens.AuthenticatedEncryptionProvider";
+        internal const string _skipValidationOfAuthenticationTagLength = "Switch.Microsoft.IdentityModel.SkipAuthenticationTagLengthValidation"; 
+
+        /// <summary>
+        /// Mapping from algorithm to the expected authentication tag length.
+        /// </summary>
+        internal static readonly Dictionary<string, int> ExpectedAuthenticationTagBase64UrlSizeInBytes = new()
```

Comment:
The map in SymmetricSignatureProvider has a lot of overlap

Reply:
When I was chatting with Brent, he expressed that we shouldn't conflate auth tag and signature. Using "ExpectedSgnatureSizeInBytes" might be confusing. I'm happy to use it an replace this one altogether, since we may remove the AesGcm values as they don't replicate this bug, if we're okay with it being slightly confusing.

Reply:
I went ahead and used "SymmetricSignatureSizeInBytes" since a customer also mentioned it in the issue.